library/std/src/fs/tests.rs

```diff
@@ -28,6 +36,23 @@ macro_rules! check {
     };
 }
 
+#[cfg(any(
+    windows,
+    target_os = "freebsd",
+    target_os = "linux",
+    target_os = "netbsd",
+    target_vendor = "apple",
+))]
+macro_rules! check_would_block {
+    ($e:expr) => {
+        match $e {
+            Ok(_) => panic!("{} acquired lock when it should have failed", stringify!($e)),
+            Err(TryLockError::WouldBlock) => (),
+            Err(e) => panic!("{} failed with: {e}", stringify!($e)),
```

Comment:
wait.

why isn't this just [`assert_matches!`](https://doc.rust-lang.org/std/assert_matches/macro.assert_matches.html)?

Reply:
No particular reason. I was just following the style of the `check` macro above. And it did seem nice to have clear error messages to differentiate why `Ok(_)` is a failure vs an IO error.

I don't have a strong preference though. Want me to remove the `check_would_block` macro and use `assert_matches` instead?

Reply:
Ah... well, that macro dates back to before Rust 1.0. It has never been updated stylistically despite evolutions in the language which can make code like that somewhat easier to write and read.

Reply:
Ah, got it. I didn't realize it was that old! Ok, updated PR coming :)

Reply:
This was the where/when it was added: https://github.com/rust-lang/rust/blob/6bfbad937bdf578e35777d079f8dcfab49758041/src/libstd/fs.rs

Reply:
Oh, wow :sweat_smile: 

Ok, I've updated it, and Github CI is green. Hopefully I didn't miss anything else for bors' test suite.

Thanks for your patience with this!